src/interface.rs

```diff
@@ -126,15 +130,32 @@ where
     ///
     /// Most likely there was a mistake with the 2in9 busy connection
     /// //TODO: use the #cfg feature to make this compile the right way for the certain types
-    pub(crate) fn wait_until_idle(&mut self, is_busy_low: bool) {
-        //tested: worked without the delay for all tested devices
-        //self.delay_ms(1);
+    pub(crate) fn wait_until_idle(
+        &mut self,
+        is_busy_low: bool,
+        mut task: Option<(
+            &mut SPI,
+            &mut DELAY,
+            fn(&mut Self, &mut SPI, &mut DELAY) -> Result<(), SPI::Error>,
+        )>,
+    ) -> Result<(), SPI::Error> {
+        // //tested: worked without the delay for all tested devices
+        // //self.delay_ms(1);
 
+        // Some displays need special treatment (Only 7.5"V2 known so far)
+        // In those cases, a "task" is provided and run here. If note, this
+        // just busy waits for the display.
         while self.is_busy(is_busy_low) {
-            //tested: REMOVAL of DELAY: it's only waiting for the signal anyway and should continue work asap
-            //old: shorten the time? it was 100 in the beginning
-            //self.delay_ms(5);
+            match task {
+                // TODO: Ignore this error?
+                Some((ref mut spi, ref mut delay, tcb)) => tcb(self, spi, delay)?,
+                None => {}
+            }
+            // //tested: REMOVAL of DELAY: it's only waiting for the signal anyway and should continue work asap
+            // //old: shorten the time? it was 100 in the beginning
+            // //self.delay_ms(5);
```

Comment:
Since we need to insert a function here anyway wouldn't it be better to just "overwrite" the waveshare.wait_until_idle and handle everything there already (at a higher level) for these specific cases?

Reply:
If not we should add a small doc example to highlight how it can be used (or link to the 7in5 v2 impl of yours above)